sycl/test-e2e/Matrix/SG32/element_wise_all_ops_half.cpp

```diff
@@ -8,7 +8,7 @@
 // SG size = 32 is not currently supported for SYCL Joint Matrix by IGC on DG2
 // UNSUPPORTED: gpu-intel-dg2
 // REQUIRES: aspect-fp16
-// REQUIRES: aspect-ext_intel_matrix, gpu
+// REQUIRES: aspect-ext_intel_matrix
```

Comment:
half is not supported on SPR. we should query based on the type as well in the hpp file.

Reply:
I already did that here: sycl/test-e2e/Matrix/element_wise_all_ops_half_impl.hpp